src/Yaapii.Atoms/Enumerable/ItemAt.cs

```diff
@@ -163,22 +165,30 @@ public sealed class ItemAt<T> : IScalar<T>
         /// <param name="source">source enum</param>
         /// <param name="position">position of item</param>
         /// <param name="fallback">fallback func</param>
-        public ItemAt(IEnumerable<T> source, int position, IBiFunc<Exception, IEnumerable<T>, T> fallback)
+        public ItemAt(IEnumerable<T> source, int position, IBiFunc<Exception, IEnumerable<T>, T> fallback) : this(
+            new ScalarOf<T>(
+                ()=>
+                    {
+                        return new ItemAtEnumerator<T>(
+                           source.GetEnumerator(), position, fallback
+                       ).Value();
+                    }
+                )
+            )
         {
-            this._pos = position;
-            this._src = source;
-            this._fbk = fallback;
         }
 
+        public ItemAt(IScalar<T> saved)
```

Comment:
please make this ctor internal instead of public.

Reply:
why not private?

Reply:
Because then we cannot test that ctor. We can't call private ctors from other assemblies (tests are another assembly). 
Internal ctors can be called from other assemblies, if we declare it as a "friendly assembly".

Reply:
But we do not use that ctor in tests. I think it is also dangerous to call that ctor from our other assemblies.

Reply:
ctor is internal now.